yt_dlp/extractor/fptplay.py

```diff
@@ -0,0 +1,103 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import hashlib
+import time
+import urllib.parse
+
+from .common import InfoExtractor
+from ..utils import (
+    traverse_obj,
+)
+
+
+class FptplayIE(InfoExtractor):
+    _VALID_URL = r'https?://fptplay\.vn/(?P<type>(?:xem-video))/([^/]+\-(?P<id>\w+))(?:\/|$)(?:tap-(?P<episode>[^/]+)|$)'
```

Comment:
untested
```suggestion
    _VALID_URL = r'https?://fptplay\.vn/(?P<type>xem-video)/[^/]+\-(?P<id>\w+)(?:/tap-(?P<episode>[^/]+)?/?(?:[?#]|$)'
```
`type` is always `xem-media`?